backend/organization/utility/requests.py

```diff
@@ -0,0 +1,156 @@
+from typing import (Dict, Optional)
+from rest_framework import status
+from rest_framework.response import Response
+from enum import Enum
+from django.utils import timezone
+from django.db.models import Q
+
+from climateconnect_api.models import Role
+from .project import add_project_member
+from .organization import add_organization_member
+from ..models.members import MembershipRequests
+from . import MembershipTarget, RequestStatus
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+
+
+
+
+
+class MembershipRequestsManager(object):
+    """
+    Manages the project/org membership requests. 
+    2 instantiations possible: 
+    A. Either to create a new request. In that case, supply the paramn: user, membership_target, user_availability, project, organization, message
+    B. To manage an existing request, provide the request id only as a kwargs. If project or organization are added, then the class check for the consistency of the supplied id 
+    and the corresponding project/ organization. If validation fails, property validation_failed will be True and errors property will contain a list of errors.
+    """
+    def __init__(self,**kwargs):
+        
+        self.validation_failed = False
+        self.duplicate_request = False 
+        
+        self.errors = list()
+        if 'membership_request_id' not in kwargs.keys(): #new request
```

Comment:
Since the code executed in the if and the else statement is completely different I think it would be better to just make 2 different functions for the 2 different use cases. But not super sure about this.